tests/test_dmlscheduler.py

```diff
@@ -35,21 +38,52 @@ def mnist_filepath():
 class MockCommunicationManager:
     def inform(self, dummy1, dummy2):
         pass
+    def configure(self, scheduler, dataset_manager):
+        pass
 
-@pytest.fixture(scope='session')
+@pytest.fixture
 def communication_manager():
     communication_manager = MockCommunicationManager()
     return communication_manager
 
+@pytest.fixture
+def blockchain_gateway(config_manager, ipfs_client, communication_manager, dataset_manager):
+    blockchain_gateway = BlockchainGateway()
+    blockchain_gateway.configure(config_manager, communication_manager, ipfs_client, dataset_manager)
+    return blockchain_gateway
+
+@pytest.fixture
+def dataset_manager(config_manager):
+    dataset_manager = DatasetManager(config_manager)
+    dataset_manager.bootstrap()
+    return dataset_manager
 # NOTE: This is scope=function and not scope=session because of multiprocessing.
 # If it's scope=session, tests will get interrupted by each other and then the assertions
 # will fail because more jobs will be processed than they expected.
```

Comment:
This note only applies for scheduler, right? Could you make the above fixtures scope=session too?

Reply:
Thanks